sdk/provisioning/Azure.Provisioning/src/resourcemanager/ResourceManagerExtensions.cs

```diff
@@ -27,13 +27,13 @@ public static ResourceGroup AddResourceGroup(this IConstruct construct)
         }
 
         /// <summary>
-        /// Gets or adds a resource group to the construct.
+        /// Gets the resource group of the construct.
         /// </summary>
         /// <param name="construct">The construct.</param>
         /// <returns>The see <see cref="ResourceGroup"/>.</returns>
-        public static ResourceGroup GetOrAddResourceGroup(this IConstruct construct)
+        public static ResourceGroup? GetResourceGroup(this IConstruct construct)
```

Comment:
this should not return null, it should just have an anon rg similar to how we did anon sub.  The instance still exists its name is just `resourceGroup()` instead of `myRG`